### PR TITLE
Annotation consistency

### PR DESCRIPTION
**Pull request by wingdeans**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Fixes `RZ_*` and `/*<type>*/` annotation inconsistencies between headers and source files

### CODE REVIEW

File: librz/bin/bobj.c [hunk 1]
```diff
@@ -771,15 +771,15 @@ static RzList *get_sections_or_segment(RzBinObject *obj, bool is_segment) {
...
+RZ_API RZ_OWN RzList *rz_bin_object_get_sections(RzBinObject *obj) {
 	rz_return_val_if_fail(obj, NULL);
 	return get_sections_or_segment(obj, false);
 }
 
 /**
  * \brief Get list of \p RzBinSection representing only the segments of the binary object.
  */
-RZ_API RzList *rz_bin_object_get_segments(RzBinObject *obj) {
+RZ_API RZ_OWN RzList *rz_bin_object_get_segments(RzBinObject *obj) {
```
Comment (wargio) [inline librz/bin/bobj.c]:
```suggestion
RZ_API RZ_OWN RzList *rz_bin_object_get_segments(RZ_NONNULL RzBinObject *obj) {
```

Same also for the others above

File: librz/include/rz_analysis.h [hunk 2]
```diff
@@ -1595,9 +1595,9 @@ RZ_API bool rz_analysis_il_vm_setup(RzAnalysis *analysis);
 RZ_API void rz_analysis_il_vm_cleanup(RzAnalysis *analysis);
 
 /* trace */
-RZ_API RzAnalysisRzilTrace *rz_analysis_rzil_trace_new(RzAnalysis *analysis, RzAnalysisILVM *rzil);
+RZ_API RzAnalysisRzilTrace *rz_analysis_rzil_trace_new(RzAnalysis *analysis, RZ_NONNULL RzAnalysisILVM *rzil);
 RZ_API void rz_analysis_rzil_trace_free(RzAnalysisRzilTrace *trace);
-RZ_API void rz_analysis_rzil_trace_op(RzAnalysis *analysis, RzAnalysisILVM *rzil, RzAnalysisLiftedILOp op);
+RZ_API void rz_analysis_rzil_trace_op(RzAnalysis *analysis, RZ_NONNULL RzAnalysisILVM *rzil, RZ_NONNULL RzAnalysisLiftedILOp op);
```
Comment (wargio) [inline librz/include/rz_analysis.h:1600]:
```suggestion
RZ_API void rz_analysis_rzil_trace_op(RZ_NONNULL RzAnalysis *analysis, RZ_NONNULL RzAnalysisILVM *rzil, RZ_NONNULL RzAnalysisLiftedILOp op);
```

Comment (wargio) [reply librz/include/rz_analysis.h:1600]: same thing for the other modified method

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Comment by wingdeans** [comment]
Still improving the inconsistency detection script. I'll merge tomorrow

**Pull request merged**